### PR TITLE
add new view

### PR DESCRIPTION
**Pull request by kingman**
for latest event per user within last 72 hours

### CODE REVIEW

File: definitions/digital_analytics_domain/product_v1/latest_event_per_user_last_72_hours.sqlx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    'event_timestamp'
+  ]
+}
+
+SELECT
+${columns}
+FROM (
+  SELECT
+    ${columns},
+    ROW_NUMBER() OVER (PARTITION BY user_pseudo_id ORDER BY event_timestamp DESC) AS rn
```
Comment (slilichenko) [inline definitions/digital_analytics_domain/product_v1/latest_event_per_user_last_72_hours.sqlx:39]: Is user_pseudo_id truly globally unique? I don't think so, but maybe mistaken. If not - need to have another column(s) in the PARTITION BY clause.

File: definitions/digital_analytics_domain/product_v1/latest_event_per_user_last_72_hours.sqlx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  FROM
+    ${ref({schema: functions.baseSchema("ga4"), name: "event"})}
+  WHERE
+    event_timestamp > TIMESTAMP_SUB(CURRENT_TIMESTAMP(), INTERVAL 3 DAY)
+    AND event_type IN ('tagged') )
+WHERE
```
Comment (slilichenko) [inline definitions/digital_analytics_domain/product_v1/latest_event_per_user_last_72_hours.sqlx:45]: This is one (canonical) way to get the latest event. A more performant way to do that is by using ARRAY_AGG with LIMIT 1 clause and get the record using OFFSET(0).

File: definitions/digital_analytics_domain/product_v1/latest_event_per_user_last_72_hours.sqlx (hunk 1 above)
Comment (kingman) [reply definitions/digital_analytics_domain/product_v1/latest_event_per_user_last_72_hours.sqlx]: added group by user_id together with user_pseudo_id which is the only two user identifier in the data.

File: definitions/digital_analytics_domain/product_v1/latest_event_per_user_last_72_hours.sqlx (hunk 2 above)
Comment (kingman) [reply definitions/digital_analytics_domain/product_v1/latest_event_per_user_last_72_hours.sqlx:44]: changed accordingly and tests show significant reduction in 'Slot time consumed' and 'Bytes shuffled', thank you so much!

File: definitions/digital_analytics_domain/product_v1/latest_event_per_user_last_72_hours.sqlx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  schema: functions.productSchema("ga4"),
+  description: "Latest event per user within the last 72 hours window",
+  tags: ['ga4']
+}
+
+js {
+  const event_type_column = 'event_type';
+  const timestamp_column = 'event_timestamp';
+  const group_by_columns = [
+      'user_id',
```
Comment (slilichenko) [inline definitions/digital_analytics_domain/product_v1/latest_event_per_user_last_72_hours.sqlx:27]: Is it the correct grouping of these records? If that's correct, LGTM.

**Review by slilichenko — APPROVED** [review]
Please check the grouping columns to make sure they are correct. If they are - approved.

File: definitions/digital_analytics_domain/product_v1/latest_event_per_user_last_72_hours.sqlx (hunk 3 above)
Comment (kingman) [reply definitions/digital_analytics_domain/product_v1/latest_event_per_user_last_72_hours.sqlx:27]: yes, user_id is the unique customer identifier that the customer need to set, if it's not set we'll fallback to user_pseudo_id.

**Pull request merged**